File: packages/java/src/main/java/io/opensms/Lookups.java
package io.opensms;

import io.opensms.models.Lookup;

import java.util.Map;

/** Number lookups (carrier, portability, validity). Accessed as {@code client.lookups()}. */
public final class Lookups {

    private final ApiTransport t;

    Lookups(ApiTransport t) {
        this.t = t;
    }

    /**
     * @param to number in E.164.
     * @return the lookup ({@code completed}, or pending: poll {@link #get}).
     */
    public Lookup create(String to) {
        return create(to, null);
    }

    /**
     * @param to      number in E.164.
     * @param options per-call options.
     * @return the lookup.
     */
    public Lookup create(String to, RequestOptions options) {
        Map<String, Object> body = Wire.map();
        body.put("to", Wire.require(to, "to"));
        return t.json("POST", "/v1/lookup", body, ApiTransport.idempotencyKey(options),
                ApiTransport.type(Lookup.class));
    }

    /**
     * @param id lookup id.
     * @return the lookup.
     */
    public Lookup get(String id) {
        return t.get("/v1/lookup/" + Wire.id(id, "id"), null, ApiTransport.type(Lookup.class));
    }
}

File: packages/php/src/Resources/Lookups.php
<?php

declare(strict_types=1);

namespace Opensms\Resources;

use Opensms\Models;
use Opensms\Page;
use Opensms\Transport;

/**
 * The `lookups` resource: number validity, carrier and porting lookups.
 * Accessed as `$client->lookups`.
 */
final class Lookups
{
    /** @internal */
    public function __construct(private readonly Transport $transport)
    {
    }

    /**
     * Look up a number. Returns the Lookup (`state` completed, or pending).
     *
     * @param array{to: string} $params
     * @param array{idempotencyKey?: string} $options
     * @return array<string, mixed>
     */
    public function create(array $params, array $options = []): array
    {
        $body = Models::map($params, Models::LOOKUP_CREATE, ['to']);

        return Models::object($this->transport->request('POST', '/v1/lookup', [], $body, [
            'idempotent' => true,
            'idempotencyKey' => Models::idempotencyKey($options),
        ]));
    }

    /**
     * @return array<string, mixed>
     */
    public function get(string $id): array
    {
        return Models::object($this->transport->request('GET', '/v1/lookup/' . Models::pathParam($id)));
    }
}
